### PR TITLE
Copy infos from previous episode

### PR DESCRIPTION
**Pull request by alexdavey**
# Description

Currently, the infos step data from the previous time step is not copied if the environment was truncated/terminated without reset. This PR fixes that. The number of infos per episode is now always the same as the number of observations.

This was mentioned previously in issue #91.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: minari/data_collector/data_collector.py [hunk 1]
```diff
@@ -209,13 +209,19 @@ def step(
...
+                    {}, self._previous_eps_final_info
+                )
             self._new_episode = False
 
         # add step data to last episode buffer
         self._buffer[-1] = self._add_to_episode_buffer(self._buffer[-1], step_data)
 
         if step_data["terminations"] or step_data["truncations"]:
             self._previous_eps_final_obs = step_data["observations"]
+            if self._record_infos:
```
Comment (rodrigodelazcano) [inline minari/data_collector/data_collector.py:223]: Can you add a small comment on why we are saving the lat obs and info?

**Review by rodrigodelazcano — COMMENTED** [review]
LGTM, just the tests missing :). @balisujohn any suggestions?

File: minari/data_collector/data_collector.py (hunk 1 above)
Comment (rodrigodelazcano) [inline minari/data_collector/data_collector.py:223]: Can you add a small comment on why we are saving the last obs and info?

**Review by rodrigodelazcano — APPROVED** [review]

File: minari/data_collector/data_collector.py (hunk 1 above)
Comment (alexdavey) [reply minari/data_collector/data_collector.py:247]: Done. Also updated the comment above to reflect the fact that infos are also being saved.

**Comment by balisujohn** [comment]
LGTM :^) Skipping adding the test for now, because we need to add somewhat involved tests for info.

**Pull request merged**